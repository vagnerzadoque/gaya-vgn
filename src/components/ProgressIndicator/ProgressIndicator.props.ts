import { Size } from '@naturacosmeticos/natds-themes'
import { BrandTypes } from '../../brandTypes/brandTypes'

export type ProgressIndicatorSize = keyof Pick<Size, 'standard' | 'semi' | 'medium' | 'large'>
export interface ProgressIndicatorProps {
  /**
   * Describe the purpose of Progress Indicator
   * Read more [here](https://dequeuniversity.com/rules/axe/4.2/aria-progressbar-name?application=axeAPI)
   */
  ariaLabel: string

  /**
   * Specify an optional className to be added to your ProgressIndicator
   */
  className?: string
  /**
   * Specify an optional Brand to be added to your ProgressIndicator
   */
  brand?: BrandTypes

  /**
   * Show a layer behind the progress indicator
   * @default false
   */
  showBackdrop?: boolean

  /**
   * The Progress Indicator size
   * @default medium
   */
  size?: ProgressIndicatorSize

  /**
   * Optional ID for testing
   */
  testID?: string,
}
