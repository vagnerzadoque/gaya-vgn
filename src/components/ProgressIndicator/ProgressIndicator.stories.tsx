/* eslint-disable max-len */
import React from 'react'
import type { Meta, StoryObj } from '@storybook/react'
import StoryContainer from '../../helpers/StoryContainer'
import { ProgressIndicator, ProgressIndicatorProps } from '.'

const componentStatus = `
> Progress indicators express an unspecified wait time.

## Properties
| Property                      | Values                           |    Status           |
|---                            |                               ---|                  ---|
| **variant(no prop)**          | circular                         | ✅ Available        |
| **ariaLabel**                 | string                           | ✅ Available        |
| **showBackdrop**              | true/false                       | ✅ Available        |
| **size**                      | standard, semi, medium, large    | ✅ Available        |
| **Layer**                     | (available but with alternative composition)    | ✅ Available        |
| **brand**                     | avon, avon_v2, natura, natura_v2, theBodyShop, <br /> consultoriaDeBeleza, casaEestilo            | ✅ Available        |

## Technical Usages Examples
`

const meta: Meta<ProgressIndicatorProps> = {
  title: 'Components/ProgressIndicator',
  component: ProgressIndicator,
  parameters: {
    componentSubtitle: '',
    docs: { description: { component: componentStatus } }
  }
};

export default meta;
type Story = StoryObj<ProgressIndicatorProps>;

export const Playground: Story = {
  render: (args) => <ProgressIndicator {...args} />
};

export const Size: Story = {
  render: (args) => (
    <StoryContainer>
      <ProgressIndicator {...args} size="standard" />
      <ProgressIndicator {...args} size="semi" />
      <ProgressIndicator {...args} />
      <ProgressIndicator {...args} size="large" />
    </StoryContainer>
  )
}
export const Backdrop: Story = {
  args: {
    showBackdrop: true
  },
  render: (args) => (
    <div style={{
      alignItems: 'center',
      backgroundColor: '#00000099',
      display: 'flex',
      height: '100vh',
      justifyContent: 'center'
    }}
    >
      <ProgressIndicator {...args} />
    </div>
  )
}
