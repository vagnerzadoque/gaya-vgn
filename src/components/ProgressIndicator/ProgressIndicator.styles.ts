/* eslint-disable max-len */
/* eslint-disable max-lines-per-function */
import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { ProgressIndicatorProps } from './ProgressIndicator.props'
import { buildTheme } from '../../ThemeProvider'

type ProgressIndicatorStyleProps = Pick<ProgressIndicatorProps, 'size' | 'showBackdrop' | 'brand'>

const themeSelectProgress = (theme: Theme) => ({ brand }: ProgressIndicatorStyleProps) => {
  const themeSelectRD = buildTheme(brand, 'light')
  if (brand) {
    return themeSelectRD.color.primary
  }
  return theme.color.primary
}

const styles = createUseStyles((theme: Theme) => ({
  backdrop: {
    height: ({ size }: ProgressIndicatorStyleProps) => size && theme.size[size],
    width: ({ size }: ProgressIndicatorStyleProps) => size && theme.size[size],
    backgroundColor: ({ showBackdrop }: ProgressIndicatorStyleProps) => showBackdrop && theme.color.surface,
    borderRadius: '50%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: theme.size.micro
  },
  loader: {
    height: ({ size }: ProgressIndicatorStyleProps) => size && theme.size[size],
    width: ({ size }: ProgressIndicatorStyleProps) => size && theme.size[size],
    '& svg': {
      animation: '$rotate 2.7s linear infinite',
      '& circle': {
        fill: 'none',
        strokeWidth: theme.size.micro,
        animation: '$spin 1.7s linear infinite',
        strokeLinecap: 'round',
        stroke: themeSelectProgress(theme)
      }
    }
  },
  '@keyframes spin': {
    '0%': {
      strokeDasharray: '1,200',
      strokeDashoffset: 0
    },
    '50%': {
      strokeDasharray: '100,200',
      strokeDashoffset: -35
    },
    '100%': {
      strokeDasharray: '100,200',
      strokeDashoffset: -124
    }
  },
  '@keyframes rotate': {
    '0%': {
      transform: 'rotate(0deg)'
    },
    '50%': {
      transform: 'rotate(360deg)'
    },
    '100%': {
      transform: 'rotate(720deg)'
    }
  }
}))

export default styles
