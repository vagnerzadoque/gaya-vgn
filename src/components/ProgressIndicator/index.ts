import ProgressIndicator from './ProgressIndicator'

export { ProgressIndicator }
export { ProgressIndicatorProps } from './ProgressIndicator.props'
export default ProgressIndicator
