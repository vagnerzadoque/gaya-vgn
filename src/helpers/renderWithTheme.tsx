import React from 'react'
import { JssProvider, SheetsRegistry } from 'react-jss'
import themes from '@naturacosmeticos/natds-themes'
import { render } from '@testing-library/react'
import { ThemeProvider } from '../ThemeProvider'

const renderWithTheme = (component: JSX.Element) => {
  const sheets = new SheetsRegistry()

  const renderTree = render(
    <JssProvider registry={sheets}>
      <ThemeProvider theme={themes.natura.light}>
        {component}
      </ThemeProvider>
    </JssProvider>
  )

  return { styles: sheets.toString(), component: renderTree }
}

export default renderWithTheme
