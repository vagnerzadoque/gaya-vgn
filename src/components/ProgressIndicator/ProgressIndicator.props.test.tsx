import React from 'react'
import renderWithTheme from '../../helpers/renderWithTheme'
import ProgressIndicator from './ProgressIndicator'

describe('Progress Indicator', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<ProgressIndicator ariaLabel="ds-progress-indicator" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly with the given size', () => {
    const { styles, component } = renderWithTheme(<ProgressIndicator size="standard" ariaLabel="ds-progress-indicator" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should render correctly a layer behind the component', () => {
    const { styles, component } = renderWithTheme(<ProgressIndicator showBackdrop ariaLabel="ds-progress-indicator" />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
