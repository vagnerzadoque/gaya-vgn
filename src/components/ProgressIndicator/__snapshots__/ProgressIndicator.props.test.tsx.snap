// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Indicator should render correctly a layer behind the component 1`] = `
[
  ".backdrop-0-2-1 {
  display: flex;
  padding: 4px;
  align-items: center;
  border-radius: 50%;
  justify-content: center;
}
.loader-0-2-2 {}
.loader-0-2-2 svg {
  animation: keyframes-rotate-0-2-4 2.7s linear infinite;
}
.loader-0-2-2 svg circle {
  fill: none;
  animation: keyframes-spin-0-2-3 1.7s linear infinite;
  stroke-width: 4;
  stroke-linecap: round;
}
@keyframes keyframes-spin-0-2-3 {
  0% {
    stroke-dasharray: 1,200;
    stroke-dashoffset: 0;
  }
  50% {
    stroke-dasharray: 100,200;
    stroke-dashoffset: -35;
  }
  100% {
    stroke-dasharray: 100,200;
    stroke-dashoffset: -124;
  }
}
@keyframes keyframes-rotate-0-2-4 {
  0% {
    transform: rotate(0deg);
  }
  50% {
    transform: rotate(360deg);
  }
  100% {
    transform: rotate(720deg);
  }
}
  .backdrop-d0-0-2-5 {
    height: 48px;
    width: 48px;
    background-color: #ffffff;
  }
  .loader-d1-0-2-6 {
    height: 48px;
    width: 48px;
  }
  .loader-d1-0-2-6 svg {  }
  .loader-d1-0-2-6 svg circle {
    stroke: #f4ab34;
  }",
  <div>
    <div
      aria-label="ds-progress-indicator"
      class="backdrop-0-2-1 backdrop-d0-0-2-5"
      data-testid="ds-progress-indicator"
      role="progressbar"
    >
      <div
        class=" loader-0-2-2 loader-d1-0-2-6 ProgressIndicator-gaya"
      >
        <svg
          viewBox="25 25 50 50"
        >
          <circle
            cx="50"
            cy="50"
            r="20"
          />
        </svg>
      </div>
    </div>
  </div>,
]
`;

exports[`Progress Indicator should render correctly with default props 1`] = `
[
  ".backdrop-0-2-1 {
  display: flex;
  padding: 4px;
  align-items: center;
  border-radius: 50%;
  justify-content: center;
}
.loader-0-2-2 {}
.loader-0-2-2 svg {
  animation: keyframes-rotate-0-2-4 2.7s linear infinite;
}
.loader-0-2-2 svg circle {
  fill: none;
  animation: keyframes-spin-0-2-3 1.7s linear infinite;
  stroke-width: 4;
  stroke-linecap: round;
}
@keyframes keyframes-spin-0-2-3 {
  0% {
    stroke-dasharray: 1,200;
    stroke-dashoffset: 0;
  }
  50% {
    stroke-dasharray: 100,200;
    stroke-dashoffset: -35;
  }
  100% {
    stroke-dasharray: 100,200;
    stroke-dashoffset: -124;
  }
}
@keyframes keyframes-rotate-0-2-4 {
  0% {
    transform: rotate(0deg);
  }
  50% {
    transform: rotate(360deg);
  }
  100% {
    transform: rotate(720deg);
  }
}
  .backdrop-d0-0-2-5 {
    height: 48px;
    width: 48px;
  }
  .loader-d1-0-2-6 {
    height: 48px;
    width: 48px;
  }
  .loader-d1-0-2-6 svg {  }
  .loader-d1-0-2-6 svg circle {
    stroke: #f4ab34;
  }",
  <div>
    <div
      aria-label="ds-progress-indicator"
      class="backdrop-0-2-1 backdrop-d0-0-2-5"
      data-testid="ds-progress-indicator"
      role="progressbar"
    >
      <div
        class=" loader-0-2-2 loader-d1-0-2-6 ProgressIndicator-gaya"
      >
        <svg
          viewBox="25 25 50 50"
        >
          <circle
            cx="50"
            cy="50"
            r="20"
          />
        </svg>
      </div>
    </div>
  </div>,
]
`;

exports[`Progress Indicator should render correctly with the given size 1`] = `
[
  ".backdrop-0-2-1 {
  display: flex;
  padding: 4px;
  align-items: center;
  border-radius: 50%;
  justify-content: center;
}
.loader-0-2-2 {}
.loader-0-2-2 svg {
  animation: keyframes-rotate-0-2-4 2.7s linear infinite;
}
.loader-0-2-2 svg circle {
  fill: none;
  animation: keyframes-spin-0-2-3 1.7s linear infinite;
  stroke-width: 4;
  stroke-linecap: round;
}
@keyframes keyframes-spin-0-2-3 {
  0% {
    stroke-dasharray: 1,200;
    stroke-dashoffset: 0;
  }
  50% {
    stroke-dasharray: 100,200;
    stroke-dashoffset: -35;
  }
  100% {
    stroke-dasharray: 100,200;
    stroke-dashoffset: -124;
  }
}
@keyframes keyframes-rotate-0-2-4 {
  0% {
    transform: rotate(0deg);
  }
  50% {
    transform: rotate(360deg);
  }
  100% {
    transform: rotate(720deg);
  }
}
  .backdrop-d0-0-2-5 {
    height: 24px;
    width: 24px;
  }
  .loader-d1-0-2-6 {
    height: 24px;
    width: 24px;
  }
  .loader-d1-0-2-6 svg {  }
  .loader-d1-0-2-6 svg circle {
    stroke: #f4ab34;
  }",
  <div>
    <div
      aria-label="ds-progress-indicator"
      class="backdrop-0-2-1 backdrop-d0-0-2-5"
      data-testid="ds-progress-indicator"
      role="progressbar"
    >
      <div
        class=" loader-0-2-2 loader-d1-0-2-6 ProgressIndicator-gaya"
      >
        <svg
          viewBox="25 25 50 50"
        >
          <circle
            cx="50"
            cy="50"
            r="20"
          />
        </svg>
      </div>
    </div>
  </div>,
]
`;
